### PR TITLE
Restrict exported components

### PR DESCRIPTION
**Pull request by flawedworld**

### CODE REVIEW

**Comment by grote** [comment]
@flawedworld thanks for the PR!

Do you mind adding a bit of explanation for the benefit of restricting opening those activities? Maybe also compared to the downsides of not being able to open restore activity with adb anymore?

**Comment by flawedworld** [comment]
Exported components should pretty much always be restricted. It's a best practice (see CWE-926). Other applications not within the scope of ```system|signature``` should never be able to open those activities because they do not need to. Using ADB to open activities should never be done outside of a debug environment.

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -47,6 +47,16 @@
         android:name="android.permission.QUERY_ALL_PACKAGES"
         tools:ignore="QueryAllPackagesPermission" />
 
+    <!-- Permission used to open settings -->
+    <permission 
+        android:name="com.stevesoltys.seedvault.opensettings"
```
Comment (mikeNG) [inline app/src/main/AndroidManifest.xml:52]: I would suggest changing this to "com.stevesoltys.seedvault.OPEN_SETTINGS", following AOSP permission naming

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -47,6 +47,16 @@
...
+        android:protectionLevel="system|signature" />
+
+    <!-- Permission used to open backup gui -->
+    <permission 
+        android:name="com.stevesoltys.seedvault.restorebackup"
```
Comment (mikeNG) [inline app/src/main/AndroidManifest.xml:57]: Same for this, "com.stevesoltys.seedvault.RESTORE_BACKUP"

**Review by mikeNG — CHANGES_REQUESTED** [review]

**Comment by chirayudesai** [comment]
Thanks for the PR.

By default, the 'restore option' in menu isn't enabled since restoring backups to an existing system may cause some weirdness.

adb was one way around that, but not ideal.

We should add some other way to getting to the restore menu on-device, a `*#*#RESTORE#*#*` `SECRET_CODE` would work.

Although that can be done separate from this PR.

**Comment by chirayudesai** [comment]
Rolled into https://github.com/seedvault-app/seedvault/pull/319 after resolving merge conflict.

**Pull request closed without merging**